test_get_nam.sh

```diff
@@ -77,29 +98,140 @@ sed -e "s/%NULLNAMWINPREDATAPATH%/$escSCENARIODIR/" \
     -e "s/\"%NULLAPPLYRAMP%\"/$boolApplyRamp/" \
     -e "s/\"%NULLRAMPDIST%\"/$SPATIALEXTRAPOLATIONRAMPDISTANCE/" \
      < get_nam_data.pl.json \
-     | $SCRIPTDIR/NAMtoOWIRamp.pl > /dev/null
-preFile=$($SCRIPTDIR/metadata.pl --key winPrePressureFile < NAMtoOWIRamp.pl.json)
-winFile=$($SCRIPTDIR/metadata.pl --key winPreVelocityFile < NAMtoOWIRamp.pl.json)
-mv $winFile $preFile fort.22 $SCENARIODIR
-mv get_nam_data.pl.* NAMtoOWIRamp.pl.* $SCENARIODIR
+     | $SCRIPTDIR/NAMtoOWIRamp.pl > NAMtoOWIRamp.pl.json
+preFile=$($SCRIPTDIR/bin/bashJSON.pl --key winPrePressureFile < NAMtoOWIRamp.pl.json)
+winFile=$($SCRIPTDIR/bin/bashJSON.pl --key winPreVelocityFile < NAMtoOWIRamp.pl.json)
+mv fort.22 $SCENARIODIR
+mv NAMtoOWIRamp.pl.* get_nam_status.pl.* asgs_main.sh_get_nam_status.json rotatedNAM.txt select_nam_nowcast.pl.json reproject.log lambert_diag.out $SCENARIODIR
+cp get_nam_data.pl.* $SCENARIODIR  # needed for forecast
 #  f o r e c a s t
 stage=FORECAST
 SCENARIO=namforecast
 SCENARIODIR=$RUNDIR/namforecast
 if [[ ! -d $SCENARIODIR ]]; then mkdir $SCENARIODIR ; fi
 escSCENARIODIR=${SCENARIODIR////'\/'}
+sed "s/NOWCAST/FORECAST/" < get_nam_data.pl.json > get_nam_forecast.json
+if [[ $? != 0 ]]; then
+    echo "ERROR: Failed to replace NOWCAST with FORECAST in select_nam_nowcast.pl.json with sed."
+    exit
+fi
+# download the forecast data
+get_nam_data.pl < get_nam_forecast.json > get_nam_data.pl.json
+if [[ $? != 0 ]]; then
+    echo "ERROR: Failed to download NAM forecast data with get_nam_data.pl."
+    exit
+fi
+# NAMtoOWIRamp.pl writes the ascii OWI WIN/PRE files to
+# the scenario directories
+sed -e "s/%NULLNAMWINPREDATAPATH%/$escSCENARIODIR/" \
+    -e "s/%NULLNAMWINPREGRID%/$escPtFilePath/" \
+    -e "s/\"%NULLNAMAWIPGRID%\"/218/" \
+    -e "s/%NULLNAMRAWFORMAT%/grib2/" \
+    -e "s/\"%NULLVELMULT%\"/$VELOCITYMULTIPLIER/" \
```

Comment:
There seems to be some inconsistency between macros that are replaced with `"` and those that are not. Is this due to the current state of the template? This is not a blocker, just wanted to point it out.

Reply:
I think this is because I wanted the numbers to be represented as numbers in JSON, rather than being enclosed in quotation marks as if they were strings. The quotation marks are in the template (even for numbers) because I wanted the template itself to be valid JSON (assuming that is a best practice). Looking at the template now, I see that this is still not the case for the `cyclelist` field, so it is kind of a mess. Open to suggestions.